### PR TITLE
update analysis

### PR DESCRIPTION
**Pull request by y1chi-z**
Add summary, introduction, methods and result&discussion sections in notebook.

### CODE REVIEW

File: notebooks/Load_Data_and_EDA.ipynb [hunk 1]
```diff
@@ -34,18 +50,99 @@
...
+   "cell_type": "markdown",
+   "id": "a1f39a05-24c5-4a5e-a34c-830e8efeee78",
+   "metadata": {},
+   "source": [
+    "In this project, a Support Vector Classifier was built and tuned to identify mushrooms edibility. A mushroom is classied as edible or poisonous with given color, habitat, class, and others. The final classifier performed quite well on unseen test data, with a final overall accuracy of 0.99 and F2 (beta = 2) score of 0.99. Furthermore, we use confusion matrix to show the accuracy of classification poisonous or edible mushroom. The model makes 12174 correct predictions out of 12214 test observations. 17 mistakes were predicting a poisonous mushroom as eatible (false negative), while 23 mistakes where predicting a eatible mushroom as poisonous (false positive). The model’s performance shows promise for implementation, prioritizing safety by minimizing false negatives that could result in consuming poisonous mushrooms. While false positives may lead to unnecessarily discarding safe mushrooms, they pose no safety risk. Further development is needed to make this model useful. Research should focus on improving performance and analyzing cases of incorrect predictions."
```
Comment (MasonZhang-MZ) [inline notebooks/Load_Data_and_EDA.ipynb]: "**_classied_**" → "**_classified_**".
"**_eatible_**" → "**_edible_**".
"**_where_** predicting a **_eatible_** mushroom as poisonous" → "**_were_** predicting an **_edible_** mushroom as poisonous."

File: notebooks/Load_Data_and_EDA.ipynb [hunk 2]
```diff
@@ -34,18 +50,99 @@
...
...s are dropped, because imputing a variable that has a significant proportion of missing data might introduce too much noise or bias, making it unreliable. Data was splitted with 80% being partitioned into the training set and 20% being partitioned into the test set. Three classification models including Support Vector Classifier (SVC), K-Nearest Neighbors (KNN), and Logistic Regression are used to predict whether a mushroom is edible or poisonous. The fine tuned Support Vector Classifier has the best overall performance. The hyperparameter was chosen using 5-fold cross validation with F2 score as the classification metric. Beta was chosen to be set to 2 for the F2 score to increase the weight on recall during fitting because redicting a mushroom to be edible when it is in fact poisonous could have severe health consequences. Therefore the goal is to prioritize the minimization of false negative. The Python programming language (Van Rossum and Drake 2009) and the following Python packages were used to perform the analysis: Matplotlib (Hunter, 2007), Pandas (McKinney, 2010), Scikit-learn (Pedregosa et al., 2011), NumPy (Harris et al., 2020), SciPy (Virtanen et al., 2020), UCIMLRepo."
```
Comment (MasonZhang-MZ) [inline notebooks/Load_Data_and_EDA.ipynb]: "**_vairables_**" → "**_variables_**."
"**_redicting_**" → "**_predicting_**."
"minimization of false **_negative_**" → "minimization of false **_negatives_**."

**Comment by hankunxiao** [comment]
Hi Essie,

Thank you for your great work! I really appreciate your analysis and writing. I have just a few suggestions:

1. It seems that your work is not based on the latest version of the main branch. Could you please rebase your work on commit a75ee46809d3addfea7c349fedf2cbccd37b8b78 in the main branch and resolve any potential conflicts?
2. In the Summary section, it would be more compelling to include percentages in the output. For example: “The model makes 12,174 correct predictions out of 12,214 test observations **(xx%)**.”
3. To make the hyperparameters more consistent, could we use the Greek letter “Beta” instead of the text format for all the references?
4. One last thing—would you mind updating my name in the author section to “Hankun” instead of my preferred name?

Thank you again for your efforts, and let me know if you have any questions or need assistance with these changes!

File: notebooks/Load_Data_and_EDA.ipynb [hunk 3]
```diff
@@ -1267,7 +1432,7 @@
    "id": "6c8c2497-cce7-4473-9604-2fee9235914f",
    "metadata": {},
    "source": [
-    "The tuned SVC seems to perform very well on the training set, giving an $F_{\\beta}$ score of about 0.997 $\\pm$ 0.0002."
+    "After tunning the hyperparameter, the Logistice Regression model has the mean accuracy of 0.75 and mean f2 score of 0.78 on the validation set. The KNN model has the mean accuracy of 0.74 and mean f2 score of 0.75. The SVC outperforms both Logistic Regression and KNN significantly in both accuracy of 0.99 and f2 score of 0.99. Thus, SVC is the ideal choice to identify edible or poisonous mushroom (recall is the highest priority)."
```
Comment (MasonZhang-MZ) [inline notebooks/Load_Data_and_EDA.ipynb]: "**_tunning_**" → "**_tuning_**."
"**_Logistice_** Regression" → "**_Logistic_** Regression."

File: notebooks/Load_Data_and_EDA.ipynb [hunk 4]
```diff
@@ -1373,8 +1538,45 @@
    "id": "17aa63ee-0e26-4bcf-82cd-740d6493ea65",
    "metadata": {},
    "source": [
-    "The model performs very well on the test set, with an accuracy and F2-Score of over 0.99."
+    "The prediction model performed quite well on test data, with a final overall accuracy of 0.99 and F2 (beta = 2) score of 0.99. The model only makes 40 mistakes out of 12214 test samples. 17 mistakes were predicting a poisonous mushroom as eatible (false negative), while 23 mistakes where predicting a eatible mushroom as poisonous (false positive). The model’s performance is promising for implementation, as false negatives represent potential safety risks and these errors could lead to consuming poisonous mushrooms, it is minimized to protect users. On the other hand, false positives are less harmful, they may lead to discarding safe mushrooms unnecessarily but does not endanger safety."
```
Comment (MasonZhang-MZ) [inline notebooks/Load_Data_and_EDA.ipynb]: "**_eatible_**" → "**_edible_**."
"**_where_** predicting a **_eatible_** mushroom as poisonous" → "**_were_** predicting an **_edible_** mushroom as poisonous."
"**_does_** not endanger safety" → "**_do_** not endanger safety."

File: notebooks/Load_Data_and_EDA.ipynb [hunk 5]
```diff
@@ -1373,8 +1538,45 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ebd8961e-06e7-4f8d-a4a8-3745e2dffe3c",
+   "metadata": {},
+   "source": [
+    "While the overall performance of the SVC model are impressive, efforts could focus on further reducing false negatives to enhance the safety of predictions. It might be important to take a closer look at the 40 misclassified observations and identify specific features that contribute to these misclassifications. Implementing feature engineering on those features such as encoding rare categories differently can enhance the model’s power and reduce the misclassification cases. Additionally, trying other classifiers like Decision Tree and Random Forest which are less sensitive to scaling or irrelevant features might improve the prediction.\n"
```
Comment (MasonZhang-MZ) [inline notebooks/Load_Data_and_EDA.ipynb]: "**_does_** not endanger safety" → "**_do_** not endanger safety."
"40 misclassified observations and identify specific features **_that contribute_** to these misclassifications" → "40 misclassified observations to identify specific features **_contributing_** to these misclassifications."

**Review by MasonZhang-MZ — COMMENTED** [review]
Hi Essie,
Thanks for doing the write up as well as the EDA integration, just a few typo fixes suggestion.

**Pull request merged**